### PR TITLE
fix: Adds domain option to user cookies set by SDK [CW-352]

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes # (issue)
https://linear.app/chatwoot/issue/CW-352/enhancement-track-user-across-subdomains
https://github.com/chatwoot/chatwoot/issues/3870

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nithindavid** [comment]
@scmmishra I have done the basic testing with localtunnel and it seems to be working. Can you check the code once? I'll test on some real world scenarios

**Comment by nithindavid** [comment]
Tested on .chatwoot.com domain here - https://www.loom.com/share/adc53917aa434f23868aceb3a181a734

File: app/javascript/sdk/cookieHelpers.js [hunk 1]
```diff
@@ -21,3 +22,17 @@ export const computeHashForUserData = (...args) => md5(getUserString(...args));
 
 export const hasUserKeys = user =>
   REQUIRED_USER_KEYS.reduce((acc, key) => acc || !!user[key], false);
+
+export const setCookieWithDomain = (
+  name,
+  value,
+  { expires = 365, baseDomain = undefined }
+) => {
+  let cookieOptions = {
```
Comment (scmmishra) [inline app/javascript/sdk/cookieHelpers.js:31]:
```suggestion
  const cookieOptions = {
```

File: app/javascript/sdk/cookieHelpers.js [hunk 2]
```diff
@@ -21,3 +22,17 @@ export const computeHashForUserData = (...args) => md5(getUserString(...args));
 
 export const hasUserKeys = user =>
   REQUIRED_USER_KEYS.reduce((acc, key) => acc || !!user[key], false);
+
+export const setCookieWithDomain = (
```
Comment (scmmishra) [inline app/javascript/sdk/cookieHelpers.js:26]: Let's write tests for this

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Comment by scmmishra** [comment]
@nithindavid overall the code looks good to go, just a few minor suggestions

**Comment by nithindavid** [comment]
@scmmishra I have added test cases and added the code suggestion. Please take a look.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**